### PR TITLE
Release/list by tag

### PR DESCRIPTION
**Pull request by vincentcreusotfx**
Added the ability to list by tags the resources. Did a small refactor because it looked like the group feature. Added README part also.

### CODE REVIEW

File: azure.go [hunk 1]
```diff
@@ -266,9 +275,54 @@ func (ac *AzureClient) listFromResourceGroup(resourceGroup string, resourceTypes
...
+	resources := extractResourceNames(data, subscription)
+
+	return resources, nil
+}
+
+// Returns all resource with the given couple tagname, tagvalue
+func (ac *AzureClient) listByTag(tagName string, tagValue string) ([]string, error) {
+	apiVersion := "2018-05-01"
+
+	filterTypes := url.QueryEscape(fmt.Sprintf("tagName eq '%s' and tagValue eq '%s'", tagName, tagValue))
```
Comment (brian-brazil) [inline azure.go]: hmm, is there any injection protection needed here?

File: config/config.go [hunk 2]
```diff
@@ -7,14 +7,15 @@ import (
 	"strings"
 	"sync"
 
-	yaml "gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v2"
```
Comment (brian-brazil) [inline config/config.go]: Leave this as-is.

File: README.md [hunk 3]
```diff
@@ -84,6 +84,12 @@ resource_groups:
     metrics:
     - name: "CPU Credits Consumed"
 
+resource_tags:
+  - resource_tag_name: "group"
```
Comment (brian-brazil) [inline README.md]: Should there be a type option here too?

Comment (vincentcreusotfx) [reply README.md]: Actually I removed the type for 2 reasons:
- I thought that conceptually in Azure, the tag is enough to segregate the resources to monitor
- the $filter thing in the API doesn't seem to handle correctly the types with several or and the tagName and tagValue part

File: config/config.go (hunk 2 above)
Comment (vincentcreusotfx) [reply config/config.go]: Ok

File: azure.go (hunk 1 above)
Comment (vincentcreusotfx) [reply azure.go]: Yeah ... the documentation for example for names say
https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-using-tags
Tag names can't contain these characters: <, >, %, &, \, ?, /
So I could at least remove all of these characters from the tagname string

Comment (brian-brazil) [reply azure.go]: I'm thinking more single quotes

Comment (vincentcreusotfx) [reply azure.go]: Done like you said

File: azure.go [hunk 4]
```diff
@@ -321,6 +322,12 @@ func (ac *AzureClient) listByTag(tagName string, tagValue string) ([]string, err
 	return resources, nil
 }
 
+func secureString(value string) string {
+	securedValue := strings.ReplaceAll(value, "'", "")
```
Comment (brian-brazil) [inline azure.go]: This will silently corrupt some data, you should escape it.

**Comment by brian-brazil** [comment]
This looks good, but there's conflicts.

**Comment by vincentcreusotfx** [comment]
> This looks good, but there's conflicts.

Fixed!

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**